src/codeyCommand.ts

```diff
@@ -252,12 +247,14 @@ export class CodeyCommand extends SapphireCommand {
       }
     } catch (e) {
       console.log(e);
-      return await message.channel.send(commandDetails.messageIfFailure!);
+      return await message.channel.send(commandDetails.messageIfFailure ?? defaultBackendErrorMessage);
```

Comment:
Looking back, what was wrong with the previous implementation (i.e. before `messageIfFailure? = 'Codey backend error - contact a mod for assistance';` was replaced with `messageIfFailure?: string;`)?

Reply:
The compiler was giving errors when I tried the previous implementation, so I figured I would use this instead.

Reply:
What about keeping the previous implementation (for the most part) and replacing
```ts
const pingCommandDetails: CodeyCommandDetails = {
  name: 'ping',
  aliases: ['pong'],
  description: 'Ping the bot to see if it is alive.',
  detailedDescription: `**Examples:**
    \`${container.botPrefix}ping\`
    \`${container.botPrefix}pong\``,

  isCommandResponseEphemeral: true,
  messageWhenExecutingCommand: 'Ping?',
  messageIfFailure: 'Failed to receive ping.',
  executeCommand: executeCommand,
  codeyCommandResponseType: CodeyCommandResponseType.STRING,

  options: [],
  subcommandDetails: {}
};
```
with
```ts
const pingCommandDetails = new CodeyCommandDetails();
pingCommandDetails.name = 'ping';
pingCommandDetails.aliases = ['pong'];
pingCommandDetails.description = 'Ping the bot to see if it is alive.';
pingCommandDetails.detailedDescription = `**Examples:**
\`${container.botPrefix}ping\`
\`${container.botPrefix}pong\``;
pingCommandDetails.messageWhenExecutingCommand = 'Ping?';
pingCommandDetails.messageIfFailure = 'Failed to receive ping.';
pingCommandDetails.executeCommand = executeCommand;
```
?
This would lead to the default values of `CodeyCommandDetails` being used. Also, in this way, I do not think that the fields with default values need to be optional anymore? If you agree, then similar changes can be done for the command details of the other commands.

Reply:
Personally, I prefer the first implementation. I think it's clearer and more compact. Also, switching to the second implementation would be a lot of refactoring as well.